protocol/igmp.go

```diff
@@ -293,7 +293,7 @@ func (p *IGMPv3GroupRecord) UnmarshalBinary(data []byte) error {
 	n += 2
 	p.MulticastAddress = data[n : n+4]
 	n += 4
-	if len(data) < int(8+p.NumberOfSources*4)+int(p.AuxDataLen) {
+	if len(data) < int(p.Len()) {
```

Comment:
`Len()` is `8 + uint16(p.AuxDataLen)*4 + p.NumberOfSources*4`, different from the original value, is this also a fix?

Reply:
This version is correct, and it is a fix

Reply:
Since `AuxDataLen` is actually always 0 for IGMPv3 report, so the issue is hidden.